deepinv/physics/phase_retrieval.py

```diff
@@ -0,0 +1,127 @@
+from deepinv.physics.forward import Physics, LinearPhysics
+from deepinv.physics.compressed_sensing import CompressedSensing
+import torch
+import numpy as np
+
+
+class PhaseRetrieval(Physics):
+    r"""
+    Phase Retrieval forward operator. The linear operator is defined by a :meth:`deepinv.physics.LinearPhysics` object with :math:`B`.
+
```

Comment:
It would be nice to define the full operator too. I propose:
Phase Retrieval base class corresponding to the operator

..math::

    y = |Bx|^2

The linear operator :math:`B` is defined as a :meth:`deepinv.physics.LinearPhysics` object.